core/src/test/scala/org/apache/spark/scheduler/PoolSuite.scala

```diff
@@ -292,7 +290,100 @@ class PoolSuite extends SparkFunSuite with LocalSparkContext {
     }
   }
 
-  private def verifyPool(rootPool: Pool, poolName: String, expectedInitMinShare: Int,
+  test("FIFO Scheduler should not add duplicate TaskSetManager") {
+    sc = new SparkContext(LOCAL, APP_NAME)
+    val taskScheduler = new TaskSchedulerImpl(sc)
+
+    val rootPool = new Pool("", SchedulingMode.FIFO, 0, 0)
+    val schedulableBuilder = new FIFOSchedulableBuilder(rootPool)
+
+    val taskSetManager0 = createTaskSetManager(0, 2, taskScheduler)
+    val taskSetManager1 = createTaskSetManager(1, 2, taskScheduler)
+    schedulableBuilder.addTaskSetManager(taskSetManager0, null)
+    schedulableBuilder.addTaskSetManager(taskSetManager0, null)
```

Comment:
Related fix aims to avoid adding duplicate schedulables (Pool and TSM). However, a new TSM is created for submitted TaskSet and duplicate TSM submission does not look an expected behaviour(although logic is robust enough for this case) so these test cases can be removed for clearer perspective.